src/flow/boykov_kolmogorov.jl

```diff
@@ -0,0 +1,208 @@
+"""
+Computes the max-flow/min-cut between source and target using
+Boykov-Kolmogorov algorithm.
+
+Returns the maximum flow in the network, the flow matrix and
+the partition {S,T} in the form of a vector of 1's and 2's.
+The partition vector may also contain 0's. These can be
+assigned any label (1 or 2), it is a user choice.
+
+For further details, please refer to the paper:
+
+BOYKOV, Y.; KOLMOGOROV, V., 2004. An Experimental Comparison of
+Min-Cut/Max-Flow Algorithms for Energy Minimization in Vision.
+
+Uses a default capacity of 1 when the capacity matrix isn\'t specified.
+
+Requires arguments:
+residual_graph::LightGraphs.DiGraph    # the input graph
+source::Int                            # the source vertex
+target::Int                            # the target vertex
+capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+
+Author: Júlio Hoffimann Mendes (juliohm@stanford.edu)
+"""
+
+function boykov_kolmogorov_impl{T<:Number}(
+    residual_graph::LightGraphs.DiGraph,   # the input graph
+    source::Int,                           # the source vertex
+    target::Int,                           # the target vertex
+    capacity_matrix::AbstractArray{T,2}    # edge flow capacities
+    )
+
+    n = nv(residual_graph)
+
+    flow = 0
+    flow_matrix = zeros(T, n, n)
+
+    TREE = zeros(Int, n)
+    TREE[source] = 1
+    TREE[target] = 2
+
+    PARENT = zeros(Int, n)
+
```

Comment:
Is it really necessary to lowercase this one? I am trying to be consistent with the notation in the paper just in case anyone needs to adapt the code in the future. Let me know if I should rename it and I will do.